device.go

```diff
@@ -0,0 +1,144 @@
+//
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/kata-containers/agent/pkg/uevent"
+	pb "github.com/kata-containers/agent/protocols/grpc"
+	"github.com/sirupsen/logrus"
+)
+
+type deviceDriversHandler func(device pb.Storage, spec *pb.Spec) error
+
+var deviceDriversHandlerList = map[string]deviceDriversHandler{
+	"blk": blockDeviceHandler,
```

Comment:
how about changing the driver name to "device"?

Reply:
you mean instead of "blk" ?

Reply:
Whatever the name is, it needs to be defined in the protocol instead of here.

Reply:
I think the code will not be only used for blk devices. so "blk" is not a proper name.
Even "device" doesn't provide enough information. IIUC, it is oci-style passing the devices which can be seen by the runtime/libcontainer into the container. Using "oci-device" or "device-passthrough" for the driver name instead of "blk" could be a good choice either.

Drivers are a part of the agent protocol, but I think just putting some comments on the documents is sufficient.

Reply:
Yes, comments are fine to me. The point is that the supported values need to be documented there.

The association with `pb.Storage` makes the handler block device specific. I think we can:
1. remove the association with `pb.Storage` in the handler, and rename "blk` to a more generic name
2. make sure that if a block device `dev.Path` matches `pb.Storage.Source`, it will be mounted at `pb.Storage.Mountpoint`